google/cloud/bigquery/v2/minimal/internal/job_rest_stub.cc

```diff
@@ -12,15 +12,35 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Implementation of internal interface for Bigquery V2 Job resource.
-
 #include "google/cloud/bigquery/v2/minimal/internal/job_rest_stub.h"
+#include "google/cloud/log.h"
+#include "google/cloud/status_or.h"
 
 namespace google {
 namespace cloud {
 namespace bigquery_v2_minimal_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+BigQueryJobStub::~BigQueryJobStub() = default;
+
+StatusOr<GetJobResponse> DefaultBigQueryJobStub::GetJob(
+    GetJobRequest const& request) {
+  GetJobResponse response;
+  if (request.project_id().empty()) {
+    GCP_LOG(DEBUG) << "Invalid request: " << request;
+    return Status(StatusCode::kInvalidArgument,
```

Comment:
Please use the (relatively new) factory functions in `google/cloud/internal/make_status.h`. There is also a companion macro:

```cc
  return InvalidArgumentError("Invalid GetJobRequest: project id is empty", GCP_ERROR_INFO());
```